src/components/Checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,122 @@
+import PropTypes from 'prop-types'
+import React, { forwardRef } from 'react'
+import styled, { css, down, typography } from '@xstyled/styled-components'
+
+import { Typography } from '../'
+
+const Checkbox = forwardRef(({ color, label, ...props }, ref) => {
+  return (
+    <LabelContainer color={color}>
+      <Input type='checkbox' ref={ref} {...props} />
+      <CheckIcon />
+      <Typography fontSize={3} lineHeight={3} fontWeight={0} marginLeft={6}>
+        {label}
+      </Typography>
+    </LabelContainer>
+  )
+})
+
+Checkbox.propTypes = {
+  checked: PropTypes.bool,
+  color: PropTypes.string,
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func,
+  label: PropTypes.string,
+  name: PropTypes.string
+}
+
+Checkbox.defaultProps = {
+  disabled: false,
+  label: 'Default label'
+}
+
+const LabelContainer = styled.label`
+  display: inline-flex;
+  position: relative;
+  user-select: none;
+  margin: 6;
+  margin-left: 0 !important;
+  flex-direction: row;
+  color: gray.800;
+
+  ${down(
+    'sm',
+    css`
+      display: flex;
+      margin-right: 0;
+      margin: 4;
+    `
+  )}
+
+  &
+    ~ input:checked
+    p {
+    color: gray.900;
+    ${typography}
+  }
+  ${typography}
+`
+
+const Input = styled.input`
+  position: absolute;
+  opacity: 0;
+  height: 100%;
+  width: 100%;
+
+  &:not(:disabled),
+  &:not(:disabled) + span {
+    cursor: pointer;
+  }
+
+  &:disabled,
+  &:disabled + span {
+    cursor: not-allowed;
+  }
+
+  &:checked:disabled + span {
+    background-color: gray.400;
+  }
+
+  &:checked:not(:disabled) + span {
+    background-color: blue.400;
+    border-color: blue.400;
+  }
+
+  &:disabled + span {
+    border-color: gray.400;
+  }
+
+  &:disabled ~ p {
+    color: gray.500 !important;
```

Comment:
evitar ao máximo o uso de `!important`. além disso, utilizar os "atalhos" do xstyled para facilitar a sobreposição de props ao utilizar o design system

```suggestion
    color: disabled;
```